### PR TITLE
DM-29375: Update UWS backend API.

### PR DESCRIPTION
**Pull request by ktlim**

### CODE REVIEW

File: python/lsst/dm/OCPS/ocps_csc.py [hunk 1]
```diff
@@ -110,38 +144,111 @@ def __init__(
             simulation_mode=simulation_mode,
         )
         self.cmd_execute.allow_multiple_callbacks = True
+        if hasattr(self.config, "triggers"):
```
Comment (timj) [inline python/lsst/dm/OCPS/ocps_csc.py:147]: whatever this `config` is it doesn't support `in` operator?

File: python/lsst/dm/OCPS/ocps_csc.py [hunk 2]
```diff
@@ -110,38 +144,111 @@ def __init__(
...
+        The ``data.config`` attribute is expected to contain ``-i`` options
+        beyond the default OODS input collection and ``-c`` options to
+        override configuration values.
+        """
         if self.simulation_mode == 0:
             # Real command.
+            run_options = ""
+            if hasattr(self.config, "input_collection"):
+                run_options = f"-i {self.config.input_collection}"
+            payload_env = dict(
```
Comment (timj) [inline python/lsst/dm/OCPS/ocps_csc.py:230]: I'm wondering about using pydantic at this point.

**Review by timj — APPROVED** [review]
I've not really looked at this code before but it seems plausible.

File: python/lsst/dm/OCPS/ocps_csc.py (hunk 1 above)
Comment (ktlim) [reply python/lsst/dm/OCPS/ocps_csc.py:147]: The documentation claims it's a `types.SimpleNamespace`, which does not have `in`.

File: python/lsst/dm/OCPS/ocps_csc.py (hunk 2 above)
Comment (ktlim) [reply python/lsst/dm/OCPS/ocps_csc.py:230]: Russell already has a whole JSON Schema thing going.

**Pull request merged**